dm/pkg/ha/bound.go

```diff
@@ -202,46 +202,58 @@ func GetLastSourceBounds(cli *clientv3.Client) (map[string]SourceBound, int64, e
 // if source bound is empty, will return an empty sourceBound and an empty source config
 // if source bound is not empty but sourceConfig is empty, will return an error
 // if the source bound is different for over retryNum times, will return an error.
-func GetSourceBoundConfig(cli *clientv3.Client, worker, source string) (SourceBound, *config.SourceConfig, int64, error) {
+func GetSourceBoundConfig(cli *clientv3.Client, worker, source string) (map[string]SourceBound, map[string]*config.SourceConfig, int64, error) {
 	var (
-		bound    SourceBound
-		newBound SourceBound
-		cfg      *config.SourceConfig
-		ok       bool
-		retryNum = defaultGetSourceBoundConfigRetry
-		sbm      map[string]SourceBound
+		bounds    map[string]SourceBound
+		newBounds map[string]SourceBound
+		configs   map[string]*config.SourceConfig
+		ok        bool
+		retryNum  = defaultGetSourceBoundConfigRetry
 	)
 	wbm, rev, err := GetSourceBound(cli, worker, source)
 	if err != nil {
-		return bound, cfg, 0, err
+		return nil, nil, 0, err
 	}
-	if sbm, ok = wbm[worker]; !ok {
-		return bound, cfg, rev, nil
+	if bounds, ok = wbm[worker]; !ok {
+		return nil, nil, rev, nil
 	}
-	bound = GetSourceBoundFromMap(sbm)
 
 	for retryCnt := 1; retryCnt <= retryNum; retryCnt++ {
 		txnResp, rev2, err2 := etcdutil.DoTxnWithRepeatable(cli, etcdutil.ThenOpFunc(clientv3.OpGet(common.UpstreamBoundWorkerKeyAdapter.Encode(worker), clientv3.WithPrefix()),
-			clientv3.OpGet(common.UpstreamConfigKeyAdapter.Encode(bound.Source))))
+			clientv3.OpGet(common.UpstreamConfigKeyAdapter.Path(), clientv3.WithPrefix())))
```

Comment:
yes, but now i don't hava a good idea too, maybe it is not a good idea to do such ops which is about cluster independently on the workers, it may be easy if we request to master and worker just wait or observe self part.

Reply:
i fixed it by make multi ops, we can explore better ways in feauter pr.